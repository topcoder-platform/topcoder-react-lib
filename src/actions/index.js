import authActions from './auth';
import smpActions from './smp';
import statsActions from './stats';
import termsActions from './terms';
import directActions from './direct';
import groupActions from './groups';
import errorActions from './errors';
import challengeActions from './challenge';
import profileActions from './profile';
import memberActions from './members';
import memberTaskActions from './member-tasks';
import reviewOpportunityActions from './reviewOpportunity';
import lookupActions from './lookup';

export const actions = {
  auth: authActions.auth,
  smp: smpActions.smp,
  challenge: challengeActions.challenge,
  terms: termsActions.terms,
  stats: statsActions.stats,
  direct: directActions.direct,
  groups: groupActions.groups,
  errors: errorActions.errors,
  profile: profileActions.profile,
  members: memberActions.members,
  memberTasks: memberTaskActions.memberTasks,
  reviewOpportunity: reviewOpportunityActions.reviewOpportunity,
  lookup: lookupActions.lookup,
};

export default undefined;
