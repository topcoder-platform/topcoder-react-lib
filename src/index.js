/**
 * Export the lib.
 */
import reducers, { factories as reducerFactories, factory as reducerFactory } from './reducers';

export { reducerFactories, reducerFactory };

export { reducers };

export { actions } from './actions';

export { services } from './services';

export { challenge, logger, errors, tc, time, mock } from './utils';
