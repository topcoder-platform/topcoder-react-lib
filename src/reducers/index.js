/**
 * Export reducers.
 */
import { redux } from 'topcoder-react-utils';

import auth, { factory as authFactory } from './auth';
import stats, { factory as statsFactory } from './stats';
import terms, { factory as termsFactory } from './terms';
import direct, { factory as directFactory } from './direct';
import groups, { factory as groupsFactory } from './groups';
import errors, { factory as errorsFactory } from './errors';
import challenge, { factory as challengeFactory } from './challenge';
import profile, { factory as profileFactory } from './profile';
import members, { factory as membersFactory } from './members';
import lookup, { factory as lookupFactory } from './lookup';
import memberTasks, { factory as memberTasksFactory } from './member-tasks';
import reviewOpportunity, { factory as reviewOpportunityFactory }
  from './reviewOpportunity';
import mySubmissionsManagement, { factory as mySubmissionsManagementFactory }
  from './my-submissions-management';


export function factory(options) {
  return redux.resolveReducers({
    auth: authFactory(options),
    stats: statsFactory(options),
    terms: termsFactory(options),
    direct: directFactory(options),
    groups: groupsFactory(options),
    errors: errorsFactory(options),
    challenge: challengeFactory(options),
    profile: profileFactory(options),
    lookup: lookupFactory(options),
    members: membersFactory(options),
    memberTasks: memberTasksFactory(options),
    reviewOpportunity: reviewOpportunityFactory(options),
    mySubmissionsManagement: mySubmissionsManagementFactory(options),
  });
}

export default ({
  auth,
  stats,
  terms,
  direct,
  groups,
  errors,
  challenge,
  profile,
  lookup,
  members,
  memberTasks,
  reviewOpportunity,
  mySubmissionsManagement,
});
