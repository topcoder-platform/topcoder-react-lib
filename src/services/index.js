/**
 * Export services.
 */
import * as api from './api';
import * as billing from './billing';
import * as challenge from './challenges';
import * as direct from './direct';
import * as groups from './groups';
import * as members from './members';
import * as terms from './terms';
import * as communities from './communities';
import * as reviewOpportunities from './reviewOpportunities';
import * as userSetting from './user-settings';
import * as user from './user';
import * as lookup from './lookup';

export const services = {
  api,
  billing,
  terms,
  communities,
  challenge,
  direct,
  groups,
  members,
  user,
  userSetting,
  reviewOpportunities,
  lookup,
};

export default undefined;
