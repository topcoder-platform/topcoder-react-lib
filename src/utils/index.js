/**
 * Export utils.
 */
import logger from './logger';
import * as tc from './tc';
import * as time from './time';
import * as mock from './mock';
import * as errors from './errors';
import * as filter from './challenge/filter';
import * as buckets from './challenge/buckets';
import * as sort from './challenge/sort';
import * as url from './url';
import * as submission from './submission';

const challenge = {
  filter,
  buckets,
  sort,
};

export {
  challenge,
  logger,
  tc,
  time,
  mock,
  errors,
  url,
  submission,
};
