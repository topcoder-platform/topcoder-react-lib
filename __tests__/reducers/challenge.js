import _ from 'lodash';
import { redux } from 'topcoder-react-utils';
import { mockAction } from '../../src/utils/mock';
import { actions } from '../../src';

const mockChallengeActions = {
  challenge: {
    getDetailsInit: mockAction('CHALLENGE/GET_DETAILS_INIT', '12345'),
    getDetailsDone: mockAction('CHALLENGE/GET_DETAILS_DONE', Promise.resolve({
      id: 12345,
      tag: 'v3-normalized-details',
    })),
    getDetailsDoneError: mockAction(
      'CHALLENGE/GET_DETAILS_DONE',
      { challengeId: '12345' },
      'Unknown error',
    ),
    getSubmissionsInit: mockAction('CHALLENGE/GET_SUBMISSIONS_INIT', '12345'),
    getSubmissionsDone: mockAction(
      'CHALLENGE/GET_SUBMISSIONS_DONE',
      Promise.resolve({ challengeId: '12345', submissions: [{ submissionId: '1' }] }),
    ),
    getSubmissionsDoneError: mockAction(
      'CHALLENGE/GET_SUBMISSIONS_DONE',
      { challengeId: '12345' },
      'Unknown error',
    ),
    getActiveChallengesCountDone: mockAction('CHALLENGE/GET_ACTIVE_CHALLENGES_COUNT_DONE', 5),
  },
  DETAIL_TABS: {
    DETAILS: 'details',
  },
};

_.merge(actions, mockChallengeActions);

const mockSmpActions = {
  smp: {
    deleteSubmissionDone: mockAction(
      'SMP/DELETE_SUBMISSION_DONE',
      '1',
    ),
  },
};
_.merge(actions, mockSmpActions);

jest.setMock('tc-accounts', {
  decodeToken: () => 'User object',
  isTokenExpired: () => false,
});

const reducers = require('../../src/reducers/challenge');

beforeEach(() => jest.clearAllMocks());

let reducer;
function testReducer() {
  let state;

  test('Creates expected intial state', () => {
    state = reducer(undefined, {});
    expect(state).toMatchSnapshot();
    // state = _.clone(defaultState);
  });

  test('Handles CHALLENGE/GET_DETAILS_INIT as expected', () => {
    state = reducer(state, mockChallengeActions.challenge.getDetailsInit(12345));
    expect(state).toMatchSnapshot();
  });

  test('Handles CHALLENGE/GET_DETAILS_DONE as expected', (done) => {
    redux.resolveAction(mockChallengeActions.challenge.getDetailsDone()).then((action) => {
      state = reducer(state, action);
      expect(state).toMatchSnapshot();
    }).then(done).catch(done);
  });

  test('Handles CHALLENGE/GET_DETAILS_DONE with error as expected', () => {
    state = reducer(state, mockChallengeActions.challenge.getDetailsDoneError());
    expect(state).toMatchSnapshot();
  });

  test('Handles fetchSubmissionsInit as expected', () => {
    state = reducer(state, mockChallengeActions.challenge.getSubmissionsInit());
    expect(state).toMatchSnapshot();
  });

  test('Handles fetchSubmissionsDone as expected', (done) => {
    redux.resolveAction(mockChallengeActions.challenge.getSubmissionsDone()).then((action) => {
      state = reducer(state, action);
      expect(state).toMatchSnapshot();
    }).then(done).catch(done);
  });

  test('Handles deleteSubmissionDone as expected', () => {
    state = reducer(state, mockSmpActions.smp.deleteSubmissionDone());
    expect(state).toMatchSnapshot();
  });

  test('Handles fetchSubmissionsDoneError as expected', () => {
    state = reducer(state, mockChallengeActions.challenge.getSubmissionsDoneError());
    expect(state).toMatchSnapshot();
  });

  test('Handles getActiveChallengesCountDone as expected', () => {
    state = reducer(state, mockChallengeActions.challenge.getActiveChallengesCountDone());
    expect(state).toMatchSnapshot();
  });
}

describe('Default reducer', () => {
  beforeAll(() => {
    reducer = reducers.default;
  });

  testReducer();
});

jest.clearAllMocks();
jest.resetAllMocks();

describe('Factory without http request', () => {
  beforeAll((done) => {
    reducers.factory().then((res) => {
      reducer = res;
      done();
    });
  });

  testReducer();
});

describe('Factory with server-side rendering', () => {
  beforeAll((done) => {
    reducers.factory({
      auth: {
        tokenV2: 'TcAuthTokenV2',
        tokenV3: 'TcAuthTokenV3',
      },
      challenge: {
        challengeDetails: {
          id: '12345',
          mySubmission: true,
        },
      },
    }).then((res) => {
      reducer = res;
      done();
    });
  });

  testReducer();
});

describe('Factory without server-side rendering', () => {
  beforeAll((done) => {
    reducers.factory({
      url: '/some-random-url',
    }).then((res) => {
      reducer = res;
      done();
    });
  });

  testReducer();
});
