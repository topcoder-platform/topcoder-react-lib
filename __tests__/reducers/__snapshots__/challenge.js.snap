// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default reducer Creates expected intial state 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles CHALLENGE/GET_DETAILS_DONE as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles CHALLENGE/GET_DETAILS_DONE with error as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles CHALLENGE/GET_DETAILS_INIT as expected 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "12345",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles deleteSubmissionDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [],
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles fetchSubmissionsDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles fetchSubmissionsDoneError as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles fetchSubmissionsInit as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "12345",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Default reducer Handles getActiveChallengesCountDone as expected 1`] = `
Object {
  "activeChallengesCount": 5,
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Creates expected intial state 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles CHALLENGE/GET_DETAILS_DONE as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles CHALLENGE/GET_DETAILS_DONE with error as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles CHALLENGE/GET_DETAILS_INIT as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "12345",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles deleteSubmissionDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [],
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles fetchSubmissionsDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles fetchSubmissionsDoneError as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles fetchSubmissionsInit as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "12345",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory with server-side rendering Handles getActiveChallengesCountDone as expected 1`] = `
Object {
  "activeChallengesCount": 5,
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Creates expected intial state 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles CHALLENGE/GET_DETAILS_DONE as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles CHALLENGE/GET_DETAILS_DONE with error as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles CHALLENGE/GET_DETAILS_INIT as expected 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "12345",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles deleteSubmissionDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [],
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles fetchSubmissionsDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles fetchSubmissionsDoneError as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles fetchSubmissionsInit as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "12345",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without http request Handles getActiveChallengesCountDone as expected 1`] = `
Object {
  "activeChallengesCount": 5,
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Creates expected intial state 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles CHALLENGE/GET_DETAILS_DONE as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles CHALLENGE/GET_DETAILS_DONE with error as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles CHALLENGE/GET_DETAILS_INIT as expected 1`] = `
Object {
  "checkpoints": null,
  "details": null,
  "fetchChallengeFailure": false,
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "12345",
  "loadingResultsForChallengeId": "",
  "mySubmissions": Object {},
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles deleteSubmissionDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [],
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles fetchSubmissionsDone as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "12345",
    "v2": Array [
      Object {
        "submissionId": "1",
      },
    ],
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles fetchSubmissionsDoneError as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles fetchSubmissionsInit as expected 1`] = `
Object {
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "12345",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {},
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;

exports[`Factory without server-side rendering Handles getActiveChallengesCountDone as expected 1`] = `
Object {
  "activeChallengesCount": 5,
  "checkpoints": null,
  "details": Object {
    "id": 12345,
    "tag": "v3-normalized-details",
  },
  "fetchChallengeFailure": "Unknown error",
  "loadingCheckpoints": false,
  "loadingDetailsForChallengeId": "",
  "loadingResultsForChallengeId": "",
  "loadingSubmissionsForChallengeId": "",
  "mySubmissions": Object {
    "challengeId": "",
    "v2": null,
  },
  "mySubmissionsManagement": Object {
    "deletingSubmission": false,
  },
  "registering": false,
  "results": null,
  "resultsLoadedForChallengeId": "",
  "unregistering": false,
  "updatingChallengeUuid": "",
}
`;
